pkgs/tools/nix/cached-nix-shell/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitHub, openssl, pkgconfig, ronn, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cached-nix-shell";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "xzfc";
+    repo = pname;
+    rev = "v" + version;
+    sha256 = "0j39mqhvjrqkcx4yfaxlak4832bx9n2w6a5j0gfnq1a8spp5fj8a";
+  };
+
+  cargoSha256 = "0cncv037sg1phh4if9zmf6gpnhrh49vkizr7xnrj8d3070xyylpr";
+
+  nativeBuildInputs = [ ronn ];
+  buildInputs = [ openssl ];
+
+  postBuild = ''
+    ronn -r cached-nix-shell.1.md
+  '';
+
+  postInstall = ''
+    mkdir -p $out/lib $out/share/cached-nix-shell $out/share/man/man1 $out/var/empty
+    cp target/release/build/cached-nix-shell-*/out/trace-nix.so $out/lib
+    cp rcfile.sh $out/share/cached-nix-shell/rcfile.sh
+    cp cached-nix-shell.1 $out/share/man/man1
+  '';
+
+  CNS_IN_NIX_BUILD = "1";
+
+  meta = with stdenv.lib; {
+    description = "Instant startup time for nix-shell";
+    homepage = "https://github.com/xzfc/cached-nix-shell";
+    license = with licenses; [ unlicense /* or */ mit ];
```

Comment:
No big deal but why do you have 2 license files? :)

Reply:
I prefer to publish it as a public domain, so I've chosen the Unlicense. However, perhaps the Unlicense might not be well-accepted by everyone, therefore I've added MIT as a fallback.

See also:
* https://github.com/BurntSushi/ripgrep/blob/master/FAQ.md#how-is-ripgrep-licensed
* https://opensource.stackexchange.com/a/5602